### ISSUE TITLE
Feature: Support homeserver without .well-known URL

### ISSUE THREAD

**Issue by bykof** [Awesome-Technologies/synapse-admin#16]
I am running the newest version of synapse-admin. 
After I click on "Sign in" on the landing page I get the error: 
```
Cannot read property 'm.homeserver' of undefined
```

When I try a wrong password it shows me correctly: 
```
Authentication failed, please retry
```

**Comment by awesome-manuel** [comment]
Might be related to #8

**Comment by teutat3s** [comment]
I can not reproduce this, could you please add more detail about what you enter before the error occurs so I can try to reproduce it?

**Comment by bykof** [comment]
I start the server via docker. 
The suggested link is not accessable, it says: `.well-known not available`
I enter my homeserver url, username and password, then this error shows up at the bottom of the page in a red alert box.

**Comment by teutat3s** [comment]
How did you setup your matrix server? Did you use https://github.com/spantaleev/matrix-docker-ansible-deploy ?

If so, please read https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/configuring-well-known.md

**Comment by sm2x** [comment]
Downloaded today Mar 24, and installed but I am also experiencing the same problem of bykof.

![undefined-m-homeserver-during-sign-in](https://user-images.githubusercontent.com/6595333/77375997-173d9080-6daa-11ea-9d9b-ae9cbaefe042.png)

and here is my well-known JSON

![dot-well-known-matrix-client](https://user-images.githubusercontent.com/6595333/77376024-30464180-6daa-11ea-9eb1-751b711ee197.png)

**Comment by dklimpel** [comment]
The login form gets the information from sever answer.
See: https://github.com/Awesome-Technologies/synapse-admin/issues/7#issuecomment-584330315
There is also an `well-known`. The login form does not a request to `/.well-known/matrix/client`.

**Comment by awesome-manuel** [comment]
Actually the `/.well-known/matrix/client` should only be used when no Homeserver is given, but instead the user enters a fully qualified user-id like `@foo:example.com`.

**Comment by dklimpel** [comment]
My challenge is: Admin API `_synapse` is for security reasons not on same url/port like normal `_matrix` API. When I enter an IP or url at login screen, it will be overwritten by server response url.

**Comment by awesome-manuel** [comment]
Right, but this is a different issue. It is not related to the discovery of the server using the `/.well-known/matrix/client` endpoint.

**Comment by awesome-manuel** [comment]
Fixed in 2d0ce50444e328eb2a678306294628d7423e1e79

**Comment by awesome-manuel** [comment]
See also 50b770a3122c4ecbeb7b858087679b6d2de03c19
